packages/adapter-netlify/index.js

```diff
@@ -1,83 +1,158 @@
+ * } & toml.JsonMap} NetlifyConfig
  */
 
+const files = fileURLToPath(new URL('./files', import.meta.url));
+
 /** @type {import('.')} */
-export default function (options) {
+export default function ({ split = false } = {}) {
 	return {
 		name: '@sveltejs/adapter-netlify',
 
-		async adapt({ utils }) {
-			// "build" is the default publish directory when Netlify detects SvelteKit
-			const publish = get_publish_directory(utils) || 'build';
-
-			utils.log.minor(`Publishing to "${publish}"`);
-
-			utils.rimraf(publish);
+		async adapt(builder) {
+			const netlify_config = get_netlify_config();
 
-			const files = fileURLToPath(new URL('./files', import.meta.url));
-
-			utils.log.minor('Generating serverless function...');
-			utils.copy(join(files, 'entry.js'), '.svelte-kit/netlify/entry.js');
-
-			/** @type {BuildOptions} */
-			const default_options = {
-				entryPoints: ['.svelte-kit/netlify/entry.js'],
-				// Any functions in ".netlify/functions-internal" are bundled in addition to user-defined Netlify functions.
-				// See https://github.com/netlify/build/pull/3213 for more details
-				outfile: '.netlify/functions-internal/__render.js',
-				bundle: true,
-				inject: [join(files, 'shims.js')],
-				platform: 'node'
-			};
+			// "build" is the default publish directory when Netlify detects SvelteKit
+			const publish = get_publish_directory(netlify_config, builder) || 'build';
 
-			const build_options =
-				options && options.esbuild ? await options.esbuild(default_options) : default_options;
+			// empty out existing build directories
+			builder.rimraf(publish);
+			builder.rimraf('.netlify/functions-internal');
+			builder.rimraf('.netlify/server');
+			builder.rimraf('.netlify/package.json');
+			builder.rimraf('.netlify/handler.js');
 
-			await esbuild.build(build_options);
+			builder.mkdirp('.netlify/functions-internal');
 
-			writeFileSync(join('.netlify', 'package.json'), JSON.stringify({ type: 'commonjs' }));
+			builder.log.minor(`Publishing to "${publish}"`);
 
-			utils.log.minor('Prerendering static pages...');
-			await utils.prerender({
+			builder.log.minor('Prerendering static pages...');
+			await builder.prerender({
 				dest: publish
 			});
 
-			utils.log.minor('Copying assets...');
-			utils.copy_static_files(publish);
-			utils.copy_client_files(publish);
-
-			utils.log.minor('Writing redirects...');
-
-			const redirectPath = join(publish, '_redirects');
-			utils.copy('_redirects', redirectPath);
-			appendFileSync(redirectPath, '\n\n/* /.netlify/functions/__render 200');
+			builder.writeServer('.netlify/server');
+
+			// for esbuild, use ESM; for zip-it-and-ship-it, use CJS
+			const esm = netlify_config?.functions?.node_bundler === 'esbuild';
+
+			/** @type {string[]} */
+			const redirects = [];
+
+			if (esm) {
+				builder.copy(`${files}/esm`, '.netlify');
+			} else {
+				// TODO might be useful if you could specify CJS/ESM as an option to writeServer
```

Comment:
I think this TODO would imply esbuild being included inside `writeServer`, which I don't like the idea of. Could we instead have an adapter option which returns the server build format to pass to the Vite build?

Reply:
we can cross that bridge if another adapter also needs CJS output. I don't think it's worth allowing opening the pandora's box of allowing adapters to affect build output just for this

Reply:
maybe just remove this TODO? it'll won't be an issue after Netlify supports ESM anyway

Reply:
yeah, that seems reasonable. removed